# GENAVi DNAnexus app

## General info
The GENAVi shiny app can only handle mouse and human RNASeq data.
It will map ENSEMBL IDs to hg38 gene symbols for human data.

Upload data must be in csv format and be raw counts i.e., no normalisation of data before upload. Thus it will only accept integer count values. It will not accept miRNA IDs. 

There is a potential issue with the report generated by 'Generate report' in the Gene expression tab as it does not reflect the uploaded data (contains mouse and human genes that not in input data). Other reports seem to function as expected i.e., they reflect the input data. 


## Build docker and save image

```
git clone https://github.com/scoughlan2/genavi-webapp.git
cd genavi-webapp/docker
docker build . -t scoughlan/genavi
docker save -o genavi.docker.gz scoughlan/genavi
```

N.B: This dockerfile pulls a docker image of the genavi shiny app and changes how genavi is deployed by the container so that the app can be seen at `localhost:3838` instead of `localhost:3838/GENAVi/`Â as is the case with the [container on dockerhub](https://hub.docker.com/r/cedarscompbio/genavi/) 

## Build web app

```
#the project on the dnanexus platform is called genavi-webapp

#First upload the docker image to the dnanexus platform as the build process uses the docker that is in the project on the platform (as the project is mounted by dx fuse in the src/genavi.sh script)

dx upload genavi.docker.gz 

#or with upload agent removig the <> and adding your auth token after --auth-token
./dnanexus-upload-agent-1.5.33-osx/ua --auth-token <> --project genavi-webapp genavi.docker.gz --do-not-compress

#Then build locally outside of the genavi directory 
dx build genavi -f 
```

## Other
[GENAVi github](https://github.com/alpreyes/GENAVi)

[GENAVi paper](https://bmcgenomics.biomedcentral.com/articles/10.1186/s12864-019-6073-7)

